src/colmap/geometry/sim3.h

```diff
@@ -60,6 +60,12 @@ struct Sim3d {
     return matrix;
   }
 
+  inline void FromMatrix(const Eigen::Matrix3x4d& matrix) {
```

Comment:
In terms of interface, it will be nicer to have
```
static Sim3d FromMatrix(const Eigen::Matrix3x4d& matrix) {...}
```

Reply:
It will be good to have a round-trip test for this new method using ToMatrix/FromMatrix and expecting near equality.

Reply:
Done, ready for review.